ios/FluentUI/Navigation/SideBar.swift

```diff
@@ -0,0 +1,284 @@
+	@objc optional func sideBar(_ sideBar: SideBar, didSelect item: TabBarItem, fromTop: Bool)
+
+	/// Called after the avatar view is tapped in the side bar.
+	@objc optional func sideBar(_ sideBar: SideBar, didActivate avatarView: AvatarView)
+}
+
+@objc(MSFSideBar)
+open class SideBar: UIView {
+	private enum Section: Int {
+		case top
+		case bottom
+	}
+
+	private struct Constants {
+		static let maxTabCount: Int = 5
+		static let viewWidth: CGFloat = 62.0
+		static let avatarViewSize: CGFloat = 30.0
+		static let topItemSpacing: CGFloat = 38.0
+		static let bottomItemSpacing: CGFloat = 28.0
+	}
+
+	@objc public weak var delegate: SideBarDelegate?
+
+	private var avatarViewGestureRecognizer: UITapGestureRecognizer?
+	@objc open var avatarView: AvatarView? {
+		willSet {
+			if let gestureRecognizer = avatarViewGestureRecognizer {
+				avatarView?.removeGestureRecognizer(gestureRecognizer)
+				avatarViewGestureRecognizer = nil
+			}
+
+			avatarView?.removeFromSuperview()
+		}
+		didSet {
+			if let view = avatarView {
+				view.translatesAutoresizingMaskIntoConstraints = false
+				addSubview(view)
+
+				let tapGesture = UITapGestureRecognizer(target: self, action: #selector(handleAvatarViewTapped))
+				avatarViewGestureRecognizer = tapGesture;
+				view.addGestureRecognizer(tapGesture)
+			}
+
+			setupLayout()
+		}
+	}
+
+	private let topStackView: UIStackView = {
+		let topStackView = UIStackView(frame: .zero)
+		topStackView.axis = .vertical
+		topStackView.distribution = .fillEqually
+		topStackView.alignment = .fill
+		topStackView.translatesAutoresizingMaskIntoConstraints = false
+		topStackView.spacing = Constants.topItemSpacing
+
+		return topStackView
+	}()
+
+	private let bottomStackView: UIStackView = {
+		let bottomStackView = UIStackView(frame: .zero)
+		bottomStackView.axis = .vertical
+		bottomStackView.distribution = .fillEqually
+		bottomStackView.alignment = .fill
+		bottomStackView.translatesAutoresizingMaskIntoConstraints = false
+		bottomStackView.spacing = Constants.bottomItemSpacing
+
+		return bottomStackView
+	}()
+
+	@objc open var bottomItems: [TabBarItem] = [] {
+		willSet {
+			willSetItems(in: .bottom)
+		}
+		didSet {
+			didSetItems(in: .bottom)
+		}
+	}
+
+	@objc open var topItems: [TabBarItem] = [] {
+		willSet {
+			willSetItems(in: .top)
+		}
+		didSet {
+			didSetItems(in: .top)
+		}
+	}
+
+	private func willSetItems(in section: Section) {
+		for subview in stackview(in: section).arrangedSubviews {
+			subview.removeFromSuperview()
+		}
+	}
+
+	private func didSetItems(in section: Section) {
+		let allItems = items(in: section)
+		let numberOfItems = allItems.count
+		if numberOfItems > Constants.maxTabCount {
+			preconditionFailure("tab bar items can't be more than \(Constants.maxTabCount)")
+		}
+
+		let stackView = stackview(in: section)
+
+		for item in allItems {
+			let tabBarItemView = TabBarItemView(item: item, showsTitle: false)
```

Comment:
Does TabBarItemView really require us to install our own gesture recognizer? That might be a bug in TabBarItemView?

Reply:
Yes, it does. This is how we handle touch events in TabBarView also.
Changing this is out of scope for this change.
Is there a process to file a bug and follow up for this?